src/utilities.jl

```diff
@@ -47,3 +47,5 @@ end
 
 _get_dense_similar(A::AbstractArray, args...) = similar(A, args...)
 _get_dense_similar(A::AbstractSparseMatrix, args...) = similar(nonzeros(A), args...)
+
+_Ginibre_ensemble(n::Int, rank::Int = n) = (randn(n, rank) + 1.0im * randn(n, rank)) / sqrt(2 * n)
```

Comment:
Why don't we simply write `randn(ComplexF64, n, rank)`?

Reply:
Moreover, do we really need the `/ sqrt(2 * n)` if we then normalize for the trace?

Reply:
The ***Ginibre ensemble*** is a set of random matrices with the entries chosen independently. 

Take full rank case (`rank = N`) as example, each entry of a $N\times N$ matrix is a complex number, with both the real and imaginary part are ***separately*** sampled from a normal distribution of mean zero and variance  $1/2N$.

The [circular law](https://en.wikipedia.org/wiki/Circular_law) states that when $N$ is large, with high probability, almost all the eigenvalues lie in the ***unit disk***. Moreover, they tend to be nearly uniformly distributed there.

From [this reference](http://dx.doi.org/10.1063/1.3595693), they explain the method to generate random density matrix from Ginibre Ensemble.

I tried `randn(ComplexF64, n, rank)` before, but it didn't give me correct Ginibre ensemble.

For the division of $\sqrt{2N}$, I keep this because we might use Ginibre ensemble somewhere else in the future (to generate other random stuffs instead of density matrix), so I thought it would be better to stick to the most original definition.